### PR TITLE
Preserve abstract keyword in signature file

### LINKED ISSUE #794: Fantomas replaces "abstract" in fsi, leading to compile errors
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKA7AhgW0QZwAdNpEwAxWWAHXRoBcBPA0gS3TsQCcsAbMAIR6YAFmAC8NCJgBGeOp2J0BmAF5g0AV3Qs6kADTgoeRHTpsA5nlRgA2oYigITqABJGza5Da6DziJBcAN0weDWQ0Oz8nAAZfKKgASXQkdgBlIhIAOQ1sfXs-ABZ8iABdfIBfOIdi13dwqG88qIDg0PrI%2BIBGKr9IAAVMc0QAdRZ4OmEmroAmaJqy50r8x163JnrIaUoeKecWkLDrDqjpnr3UxGwWOB5YdABBOgBRZIB5ADMAGTZkM6d3kLGeYVHorPZrDxoTbbXZOfZtI41CAAZj%2B-nSxEQ-EQ71gnEQAwUuA4nFhznkYWBi1BNQCdU8W1gOz%2B8MOESRYAKaKgGJI2Nx%2BM%2BsAA7lxoJhjElArBxWY7mSnBTEFSnEtnGC4RCNozmbSggd2hyAKzcyC8rE4vGIACqBGYnHFkvQ0tlLHl3IBPCBUQWqppzS1DJhLP1CPZ8TAADZTeb%2BVaAMJCPB4eN3OScDTQOh4hUQT3evy%2BiBqpwa-yBqE6hWsw0RgDsMYyFoFiAAspdpFxc2B88qfSDlnr6ZXg3rWmzbByAByNzFx-Gp27obu9lXF-2rYdQKsh8e1%2BIATlnJHu7xJqew2Ew3aVa7AJeqAa30KZ1dDE%2BOfk6sQ5Zqb88QC4rhud0OVXftqUHJ91iDV9dwNREI06bpf3NU8SSA64mVAiNbwgv0oM3GCRzgscEPDGZTSSFI6FedAABVOAYUYJhXQE%2B0LAd1SHYjt1HAM90QrpUVQpt7k4WAtHgAARRAeBYK4SRvDMOOcIt7w3cFn0aeCw0nJCuV-VtMAADwSd56OERB0CeL1AOEPE6FGcZJm5Ao5nw9dCK03ivHYN9BIoqJOhNIzTKSd4WBM157UwbNOCeEyCHxZM3WXbkjQ8zjIO46DIQafzdI-DlOmjMKTIAJUQOBOHgZzWLcrK1K40sePyvyfDIvTP2cToG3K8SFAYV5OG%2BOR6tcjl3LvB8wDLWpfJ0rriqQmdys%2BEx%2BDYeALAm7tps8jTvM1Z8d2W-dgqPIyNB4Mx5PQLFbmgABrfgFBekw8Do1IcEQRccnSsD2JmzSTt8s6BPI-T4lmU1MkQYV7otOhRWs%2Bj1lkyLtDlB5knbbBO04KwPWBw7Zvmulwf41ZAuhk4UIjM15GuOhWwQX4gbsmbDBKOIvDwMg8BYaxb3KIA)

In an FSI, Fantomas replaces `abstract Foo` with `member Foo`. This can cause a compile error with message "The representation of this type is hidden by the signature."

#### Code

```fsharp
namespace Foo

type internal Blah =
  abstract Baz : unit
```
            

#### Result

```fsharp
namespace Foo

type internal Blah =
    member Baz: unit

```
            
#### Options

Fantomas Next - 4.0.0-alpha-001-1/1/1990

| Name | Value |
| ---- | ----- |
| `IndentSpaceNum` | `4` |
| `PageWidth` | `120` |
| `SemicolonAtEndOfLine` | `false` |
| `SpaceBeforeParameter` | `true` |
| `SpaceBeforeLowercaseInvocation` | `true` |
| `SpaceBeforeUppercaseInvocation` | `false` |
| `SpaceBeforeClassConstructor` | `false` |
| `SpaceBeforeMember` | `false` |
| `SpaceBeforeColon` | `false` |
| `SpaceAfterComma` | `true` |
| `SpaceBeforeSemicolon` | `false` |
| `SpaceAfterSemicolon` | `true` |
| `IndentOnTryWith` | `false` |
| `SpaceAroundDelimiter` | `true` |
| `MaxIfThenElseShortWidth` | `40` |
| `MaxInfixOperatorExpression` | `50` |
| `MaxRecordWidth` | `40` |
| `MaxArrayOrListWidth` | `40` |
| `MaxLetBindingWidth` | `40` |
| `MultilineBlockBracketsOnSameColumn` | `false` |
| `NewlineBetweenTypeDefinitionAndMembers` | `false` |
| `StrictMode` | `false` |

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #794

### CODE REVIEW

**Comment by nojaf** [comment]
Trying to add @JackMatusiewiczGR as a reviewer as well.
Not sure why I can't select your username.

File: src/Fantomas/CodePrinter.fs [hunk 1]
```diff
@@ -2151,8 +2154,10 @@ and genMemberSig astContext node =
             | [] -> sepColon
             | _ -> sepColonWithSpacesFixed
 
+        let ziggy = mf
```
Comment (Smaug123) [inline src/Fantomas/CodePrinter.fs:2157]: Think you didn't mean to keep this here

Comment (nojaf) [reply src/Fantomas/CodePrinter.fs:2157]: Good catch, I tend to add stuff like that to have a better debug experience in Rider.
Shouldn't be in there at all ;)

**Comment by nojaf** [comment]
@Smaug123 added an additional test for implementation files as well.

**Pull request merged**